public/apps/account/account-app.tsx

```diff
@@ -30,6 +43,35 @@ export async function setupTopNavButton(coreStart: CoreStart, config: ClientConf
         coreStart.http.basePath.serverBasePath + CUSTOM_ERROR_PAGE_URI + ERROR_MISSING_ROLE_PATH;
     }
 
+    let tenant = accountInfo.user_requested_tenant;
+    let shouldShowTenantPanel = true;
+
+    if (tenantSpecifiedInUrl() || getShouldShowTenantPopup() === false) {
+      shouldShowTenantPanel = false;
+    } else {
+      // Switch to previous tenant based on localStorage, it may fail due to
+      // 1) Localstorage is disabled; 2) Request failed
+      try {
+        const savedTenant = getSavedTenant();
+        if (savedTenant !== null) {
+          if (savedTenant === tenant) {
+            shouldShowTenantPanel = false;
+          } else {
+            await selectTenant(coreStart.http, {
+              username: accountInfo.user_name,
+              tenant: savedTenant,
+            });
+            tenant = savedTenant;
+            shouldShowTenantPanel = false;
+          }
+        }
+      } catch (e) {
+        console.log(`Failed to switched to ${tenant} tenant.`);
```

Comment:
Shall we log e as well for troubleshooting?

Reply:
Updated with lib function `constructErrorMessageAndLog`.

Reply:
nitpicking laungauge here, the console message should read `Failed to switch to...` (remove `ed` in `switched`) 

```suggestion
        console.log(`Failed to switch to ${tenant} tenant.`);
```

Reply:
@kgcreative thanks, updated.